### DISCUSSION TITLE
How do I run Interstellar on a specific port?

### DISCUSSION THREAD

**Discussion by samer2011** [UseInterstellar/Interstellar#469] [Q&A] [answered]
I am trying to run an Interstellar on a specific port so I can forward it to Ngrok. Can someone help me? I tried npm run start, but it gave me this error:

> in@5.1.3 start
> node index.js

/home/samer/interstellar/Interstellar-main/node_modules/@tomphttp/bare-server-node/dist/createServer.js:42
    init.logErrors ??= false;
                    ^

SyntaxError: Unexpected token '?'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at ModuleWrap.<anonymous> (internal/modules/esm/translators.js:188:29)
    at ModuleJob.run (internal/modules/esm/module_job.js:145:37)
    at async Loader.import (internal/modules/esm/loader.js:182:24)
    at async Object.loadESM (internal/process/esm_loader.js:68:5)
npm ERR! code 1
npm ERR! path /home/samer/interstellar/Interstellar-main
npm ERR! command failed
npm ERR! command sh -c node index.js

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/samer/.npm/_logs/2024-02-20T21_47_16_382Z-debug.log

How can I fix this?

**Comment by GooseGooseDuck198** [comment]
1. Run `npm i`
2. Run `$env:PORT=<PORT-ID:INT>; npm run start`
3. Now forward the port you defined in the previous step (see: port forwarding)
4. Set port visibility to public (see: port visibility)

This *should* work!

NOTICE: I am not a professional, the information I gave you was obtained from a <5> minute google search and documentation reading, take this information with an *entire container of* grains of salt

**Reply by samer2011** [reply]
Thank you for the reply, @GooseGooseDuck198. I will show you how I tried.

So I used the "cat" commend on the index.js file and changed the port to 6969. Then, I used your command, so env:PORT=6969; npm run start, and I will just show you what happened.

<img width="542" alt="Screenshot 2024-02-21 151539" src="https://github.com/InterstellarNetwork/Interstellar/assets/160674984/da800336-ad44-421b-b514-3f92ca1accf1">

Now how can I fix that?

**Reply by GooseGooseDuck198** [reply]
Uuuhhh...

I'm not sure.

Check the file it mentions and see if there is a stynax error or typo

**Reply by samer2011** [reply]
Yeah i just figured it out like 5 minutes ago.

**Reply by 0mhx** [reply, marked as answer]
Your nodejs version is old

run `sudo apt uninstall node*` and then run the following:
```
sudo su
curl -fsSL https://deb.nodesource.com/setup_current.x | bash - &&\
apt-get install -y nodejs
```
and try again
